### PR TITLE
Add test of scrolling competition

### PR DESCRIPTION
**Pull request by dkwingsmt**
## Description

This PR adds a test to ensure that no exception will be thrown when a `PointerSignalEvent` is listened to by multiple `Scrollable`s. This used to be the case that `PointerSignalResolver` assumes that all events registered during a pointer event are the same object, but `GestureBinding` transforms each pointer event into a different object before dispatching, which has been fixed by https://github.com/flutter/flutter/pull/48149.

## Related Issues
- The issue has been fixed by https://github.com/flutter/flutter/pull/48149

## Tests

I added the following tests:

- "Scroll pointer signals are handled when there is competion"

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by dkwingsmt** [comment]
/cc @mehmetf I found out that the problem has been around for a long time (since a PR back in May if my analysis is correct). I wonder why it hasn't been brought up earlier.
Do we have issues of this?

File: packages/flutter/lib/src/gestures/pointer_signal_resolver.dart [hunk 1]
```diff
@@ -27,13 +27,14 @@ class PointerSignalResolver {
 
   /// Registers interest in handling [event].
   void register(PointerSignalEvent event, PointerSignalResolvedCallback callback) {
-    assert(event != null);
+    final PointerSignalEvent original = event.original as PointerSignalEvent;
+    assert(original != null);
```
Comment (chunhtai) [inline packages/flutter/lib/src/gestures/pointer_signal_resolver.dart]: Can we assume original will always not null?

**Review by chunhtai — COMMENTED** [review]
Is there an issue associate with this pr?

**Comment by goderbauer** [comment]
Looks like cirrus is unhappy with this change.

Also, it appears to be fixing the same issues as https://github.com/flutter/flutter/pull/48149. 
/cc @mdebbar

**Comment by mdebbar** [comment]
Sorry @dkwingsmt, I didn't know about this PR until now. Even though the issue is fixed in master, I think the additional tests you wrote are still valuable to have.

**Review by mdebbar — APPROVED** [review]

**Comment by dkwingsmt** [comment]
@mdebbar No worries, and I also think the additional test should still be submitted. Let me sync the PR and merge it. Thanks for the approval.

**Pull request merged**